### PR TITLE
Refactor media queries

### PR DESCRIPTION
**Pull request by matt-bernhardt**
### Why these changes are being introduced

Initially we tried to use the new media query syntax for specifying
screen widths, but that ran afoul of slightly older mobile devices,
because Safari only added support for the syntax in March 2023.

Without that support, phones were trying to render the site in desktop
mode.

### How this addresses that need

This converts the media queries in the theme from the newer context
syntax "(width <= ABCpx)" to the older prefix syntax "(max-width:
ABCpx)".

### Side effects of this change

We are adopting the variable names for screen widths for the rules
which are close to those break points. A few rules are set at 600px,
so I'm not going to try and dive into that use case right now to
coerce them into -sm or -md breakpoints right now.

### Relevant ticket(s)

https://mitlibraries.atlassian.net/browse/POST-187

### CODE REVIEW

File: asset/scss/_variables.scss [hunk 1]
```diff
@@ -4,3 +4,13 @@
...
+
+// ----------------------------
+// #GRID
+// ----------------------------
+
+// grid - breakpoints
+$bp-screen-sm: 480px;
+$bp-screen-md: 768px;
+$bp-screen-lg: 1024px;
+$bp-screen-xl: 1280px;
```
Comment (JPrevost) [inline asset/scss/_variables.scss:16]: It looks like `$bp-screen-sm` and `$bp-screen-md` are both used exactly once and `bp-screen-lg` and `$bp-screen-xl` are not used at all. Combined with there being a breakpoint we don't define here at all, I'm nos sure if this grid breakpoint definition is helpful or misleading.

Suggestions:
- commend out the breakpoints you suspect we'll use in the future but currently do not use
- add an additional named breakpoint here for our 600px width that we currently use

**Review by JPrevost — COMMENTED** [review]
Minor changes requested to clarify what is in use in variables.scss.

Otherwise if stakeholders have confirmed this is working as intended, :shipit:

File: asset/scss/_variables.scss (hunk 1 above)
Comment (matt-bernhardt) [reply asset/scss/_variables.scss:16]: I'll either make an attempt at moving the 600px rules to either small or medium, or I'll define a new intermediate size. The large and xl sizes will be commented out.

This section was a copy-pasta from the style library, but I see logic in not leaving it as-is.

**Comment by matt-bernhardt** [comment]
I've pushed a commit to address your comment, @JPrevost  - most of the rules have been updated to take effect at the medium screen size (768, rather than 600). One rule remains, which is the subject of a coming ticket to try and optimize the header size on mobile screens. I'd rather deal with this breakpoint in that ticket - https://mitlibraries.atlassian.net/browse/POST-168 . I've added this as a success criteria there.

I also see that there's now a merge conflict on this PR, because the other PR has landed. The resolution is straightforward, but I didn't want to rebase without checking in with you beforehand.

**Review by JPrevost — APPROVED** [review]

**Pull request merged**